feat: add ability to output candidate releases

niieani opened the pull request:
For the purposes of releasing, it is useful to just get the information of the packages that are about to be released.
If `only: 'list-candidate-releases'` is set, only output information about which packages would be released. 

It provides `releases_pending` and `paths_to_release` as the outputs.

In order to enable this new feature, I've added a new input called `only` that allows running only the desired functionality (`'create-github-releases' | 'list-candidate-releases' | 'update-pull-requests'`). 
It is not a breaking change, and is backwards compatible with the `skipGitHubRelease` and `skipGitHubPullRequest` inputs.

niieani commented:
Hi @chingor13! Any chance you could have a look? Open to feedback if you want any changes. 
It would be really great if we could land a way to only output releases.

🙇

le-yams commented:
Hi @niieani ,

I would be interested to also have a `list-pull-requests` feature for the `only` parameter.

Something like:

```ts
export async function main() {
  ...
  if (inputs.only === 'list-pull-requests') {
    core.debug('Listing pending pull requests');
    outputCandidatePRs(await manifest.buildPullRequests());
  }
  ...
}
...
function outputCandidatePRs(prs: ReleasePullRequest[]) {
  prs = prs.filter(pr => pr !== undefined);
  core.setOutput('prs_pending', prs.length > 0);
  if (prs.length) {
    core.setOutput('pr', prs[0]);
    core.setOutput('prs', JSON.stringify(prs));
  }
}
```
So in this case the pr and prs output would be of type [ReleasePullRequest  object](https://github.com/googleapis/release-please/blob/main/src/release-pull-request.ts#L20) instead of [PullRequest object](https://github.com/googleapis/release-please/blob/main/src/pull-request.ts#L15).

What do you think about it?
🙇

niieani commented:
Hi @le-yams ! I think that's a good addition, however note that I've been trying to get this merged since October 2024.
I do hope @chingor13 can see this at some point in time, and review / give feedback / merge.

chingor13 reviewed (changes requested):
Thanks for this.

I think I'd prefer a `dry-run` boolean option to mirror what the CLI does rather than maintaining a list of actions.

niieani committed:
feat: add dry-run

adds ability to output candidate releases, and candidate PRs without actually releasing or updating PRs

niieani commented:
Thank you for taking a look @chingor13.
Made the necessary changes and added PR dry-run too based on @le-yams's suggestion.

niieani commented:
Sorry to ping you - I was wondering if there's any chance for a merge&release given the code updated based on your review? 🙇 Much appreciated!

niieani commented:
@ferrarimarco rebased!

niieani committed:
build: update bundled action for dry-run

Motivation: branch was rebased onto the current PR base and the action entrypoint must match the dry-run source changes.
What was done: regenerated the checked-in ncc bundle after resolving the rebase conflicts.
Why this resolves it: GitHub Actions executes dist/index.js, so the packaged action now includes the dry-run behavior from src/index.ts.
